Sources/NIO/Bootstrap.swift

```diff
@@ -649,6 +794,57 @@ public final class ClientBootstrap: NIOClientTCPBootstrapProtocol {
             return eventLoop.submit(setupChannel).flatMap { $0 }
         }
     }
+    
```

Comment:
we're missing support for the "universal bootstrap": `NIOClientTCPBootstrap`.

Reply:
Will changing this break downstream somewhere?

Reply:
@PeterAdams-A we can't change in a breaking way but we can add things. So we can add a new method to the protocol as long as we provide a default implementation (which goes through `.socket(...)`.

We should definitely think the universal bootstrapping through because it's pretty much the recommended way of bootstrapping client connections because it supports to pretty seamlessly switch between NIO on Sockets & NIOTS (NIO Transport Services aka. NIO on Network.framework).

Reply:
Thinking about this. CC @Lukasa / @glbrntt how were we actually thinking to map the options from the universal bootstrap to the bootstrap specific options? Sure, we can do a `.socket` _fallback_ but it shouldn't be the only way.

Should we maybe define the convenience options just once for all TCP client bootstraps? So that the universal bootstrap can then just hand them over to the underlying bootstrap which can then map them to their favourite way (`NIO` --> `.socketOption`; `NIOTS` --> Network.framework's options)

Reply:
I kinda got the impression that was the original goal. We still need a `.socket` fallback though.

Reply:
@Lukasa agreed. But correctly, we nest them in `ClientBootstrap` which suggests we'd have multiple, ie. `NIOTSConnectionBootstrap` would have its _own_ set of options, correct @PeterAdams-A ?

Reply:
As stands it would get it's own options which wouldn't work.  Would either have to pull out common TCP based options or have some mapping which would probably be horrible.

Reply:
Actually - yes, could give it's own version and map direct at that level into old style - that is probably the easiest by far.  I'll try that first.

Reply:
I have added a set of options onto NIOClientTCPBootstrap which are directly translated into the underlying Chanel options for application rather than passing through the shorthand.

Given the shorthands are unique by underlying type it would be difficult to pass through the shorthand to the underlying.

I did try putting a shorthand method and an extension implementation onto NIOClientTCPBootstrapProtocol but this didn't obviously add any benefit that I could think of given everywhere else has a direct and separate implemention and if an external user reimplements nothing would pass though to this method anyway.